flang/include/flang/Optimizer/Support/Utils.h

```diff
@@ -112,18 +120,6 @@ inline std::string mlirTypeToIntrinsicFortran(fir::FirOpBuilder &builder,
     return "INTEGER(KIND=8)";
   else if (type.isInteger(128))
     return "INTEGER(KIND=16)";
-  else if (type == fir::ComplexType::get(builder.getContext(), 2))
```

Comment:
You're removing support for fir::ComplexType in a lot of places. Why not remove the type from tablegen too?

Reply:
I wanted to fully remove it in the next patch to avoid this one from getting too big, but I am also OK with doing that here. Would you rather have all the removal done in this patch?

Reply:
Doing it in another patch is okay